### PR TITLE
Avoid machine sleeping whilst running strap

### PR DESCRIPTION
**Pull request by sihil**
This is a simple change to prevent the machine sleeping whilst the strap script is running. Hopefully simple enough to pass the litmus test in #124? It makes use of the macOS built in CLI tool `caffeinate`.

The command I've selected here uses `-i` (prevent system from idle sleeping), `-d` (prevent display from sleeping and `-w` (wait for process with PID to exit - in this case the PID of the bash shell running the script). 

There is an argument for removing `-d` but I prefer to see progress so left it in.

### CODE REVIEW

**Comment by sihil** [comment]
I'll rebase this change against `MikeMcQuaid:master`

File: bin/strap.sh [hunk 1]
```diff
@@ -3,6 +3,8 @@
 #/ Install development dependencies on macOS.
 set -e
 
+caffeinate -di -w $$ &
```
Comment (MikeMcQuaid) [inline bin/strap.sh]: Can this go under `groups | grep $Q admin || abort "Add $USER to the admin group."` (i.e. the early aborts) and have a comment referencing what behaviour we're going for here? I'd like to also propose replacing `-d` with `-s` (as it feels like a more sensible default to only do this when the user is on AC power) and removing `-i` as the display sleeping feels a bit out of scope.

Comment (sihil) [reply bin/strap.sh]: Very happy for it to be moved and think `-s` is a reasonable invocation. Have updated and added a comment for clarity.

**Comment by MikeMcQuaid** [comment]
Great work, thanks again @sihil!
